settings/memcache.settings.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/**
+ * @file
+ * Contains configuration for the Drupal 8 Memcache API and Integration and
+ * settings for using Memcached on Acquia Cloud.
+ *
+ * Note: the memcache module must be enabled before using memcache as a backend.
+ *
+ * @see: https://api.drupal.org/api/drupal/core%21core.api.php/group/cache/8.4.x
+ * @see: https://www.drupal.org/node/2754947
+ * @see: http://cgit.drupalcode.org/memcache/tree/README.txt?h=8.x-2.x
+ * @see: https://docs.acquia.com/cloud/performance/memcached
+ */
+
+use Drupal\Core\Config\BootstrapConfigStorageFactory;
+use Drupal\Core\Database\Database;
+
+if ($is_ah_env) {
+  // Note that this connects the database, so has to go after any
+  // $databases definitions. See: https://www.drupal.org/node/2766509
+  Database::setMultipleConnectionInfo($databases);
```

Comment:
Can this be removed after https://www.drupal.org/node/2766509 is resolved?

Reply:
Something about this line makes me nervous.  it feels like we should let core handle this, lest we accidentally change the database connection behavior. Do you feel that this is safe?

Reply:
In theory yes, although have not looked at the patches extensively and I am not sure [#2766509](https://www.drupal.org/node/2766509 ) takes care the case where we move locks to the memcache backend, specially  `$settings['container_yamls'][] = __DIR__ . '/memcache.yml';`

I share your concern and I would get a strong code smell reviewing this if I came across it. I mostly wanted to get your thoughts on the approach. I am not sure if there is a better/another way to check in `settings.php` if a module is enabled, which is what we need to do until the above mentioned issue of landed. Open to other suggestions. I have tested this code and it seems to work fine, but I have not tested extensively on ACE.

I think the other option is make memcache integration opt-in and just document how to configure it for environments that are known to have Drupal installed and the Memcache module enabled, and won't be reinstalled, e.g. stage and prod.

Reply:
I was just looking https://www.drupal.org/files/issues/gracefully_handle_a-2766509-49.patch and it looks like `cache.backend.null` is used in the case that the configured backend is provided by a module that is not installed yet. I think this means that performance could be potentially terrible if BLT assumes a site will use the Memcache module on ACE, but it is never actually enabled as cacheing would essentially be disabled. I am leaning more to the opt-in solution.